src/main/java/mate/project/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,24 @@
+package mate.project.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String first;
+    private String second;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.first = constraintAnnotation.first();
+        this.second = constraintAnnotation.second();
+    }
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext context) {
+        Object firstValue = new BeanWrapperImpl(object).getPropertyValue(first);
```

Comment:
The BeanWrapperImpl instances are being created twice, which is a performance overhead. Consider creating a single instance of BeanWrapperImpl and reusing it.